plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoPrivilegesCaseInsensitiveSchema.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.MaterializedRow;
+import io.trino.testing.QueryRunner;
+import org.bson.Document;
+import org.testng.annotations.Test;
+
+import java.util.Locale;
+import java.util.Optional;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.createRole;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.createUser;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.privilege;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.resource;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.role;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestMongoPrivilegesCaseInsensitiveSchema
+        extends AbstractTestQueryFramework
+{
+    private static final String TEST_USER = "testUser";
+    private static final String TEST_PASSWORD = "pass";
+    private static final String TEST_DATABASE = "DataBase";
+    private static final String TEST_ROLE = "testRole";
+    private static final String TEST_COLLECTION = "testCollection";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        AuthenticatedMongoServer mongoServer = closeAfterClass(setupMongoServer());
+        String connectionUrl = mongoServer.testUserConnectionString(TEST_DATABASE, TEST_USER, TEST_PASSWORD).getConnectionString();
+        DistributedQueryRunner queryRunner = null;
+        try {
+            queryRunner = DistributedQueryRunner.builder(testSessionBuilder()
```

Comment:
`queryRunner` should be created before `try-catch`, otherwise `closeAllSuppress` is getting a null value for `queryRunner`.

Reply:
according to 

<details>
<summary>closeAllSuppress implementation</summary>

```
    public static <T extends Throwable> T closeAllSuppress(T rootCause, AutoCloseable... closeables)
    {
        requireNonNull(rootCause, "rootCause is null");
        if (closeables == null) {
            return rootCause;
        }
        for (AutoCloseable closeable : closeables) {
            try {
                if (closeable != null) {
                    closeable.close();
                }
            }
            catch (Throwable e) {
                // Self-suppression not permitted
                if (rootCause != e) {
                    rootCause.addSuppressed(e);
                }
            }
        }
        return rootCause;
    }
```
</details>

it's not a problem. We are safe because of 
```
        if (closeables == null) {
            return rootCause;
        }
        for (AutoCloseable closeable : closeables) {
```

Reply:
I know `closeAllSuppress` is safe to pass null. But still the construct here is smelly.